user_sync/connector/directory_ldap.py

```diff
@@ -310,13 +304,16 @@ def iter_users(self, base_dn, users_filter, extended_attributes):
         user_attribute_names.extend(self.user_email_formatter.get_attribute_names())
         user_attribute_names.extend(self.user_username_formatter.get_attribute_names())
         user_attribute_names.extend(self.user_domain_formatter.get_attribute_names())
-        user_attribute_names.append(six.text_type('memberOf'))
+
+        if not self.options['two_steps_enabled']:
+            user_attribute_names.append(six.text_type('memberOf'))
```

Comment:
`memberOf` shouldn't be requested from LDAP in any case. It is extra attribute which is ignored in most of LDAP directory but fails with above error on NetIQ eDirectory. This should be simply commented out from request.

Reply:
I agree - but I think we shouldn't hardcode this anywhere - even in the conditional.  This line will break some openldap implementations.  If a user needs memberof, they can add it to the extension attributes?

Reply:
We need to create an extra config where the attribute name can be specified. I don't think it would work to expect it to be in the extension config because this field has special meaning to the dynamic group mapping functionality.

Reply:
The way I thought when I proposed the change was to separate AD from other LDAP implementations:
- AD will get memberOf because it's a default attribute of an user object
- the rest of the LDAP will go 'two steps lookup' with a special configuration since 'member' is a group object attribute that usually openLDAP has. The default ldap queries usually need a change when using openLDAP anyway.

At the time when I proposed the change I was not fully aware how the LDAP query was being formed and the change was to not risk breaking anything, but now I guess  I could just remove complectly the conditional and have the memberOf be part of the usual ldap filters in connector-ldap file.

Both ways, this should work.

Reply:
`memberOf` is a separate issue and doesn't need to be addressed in this PR. We need to fix it in a separate PR that tracks with #503. Let's get this merged, then we can make a fix for #503.